src/Drips.sol

```diff
@@ -476,14 +476,19 @@ library Drips {
                     newDefaultEnd
                 );
                 _moveDeltaRange(
-                    deltas,
+                    state.amtDeltas,
                     cycleSecs,
                     currStart,
                     currEnd,
                     newStart,
                     newEnd,
                     currRecv.amtPerSec
                 );
+                // Ensure that the user receives the updated cycles
+                uint32 startCycle = _cycleOf(newStart, cycleSecs);
+                if (state.nextReceivedCycle > startCycle) {
+                    state.nextReceivedCycle = startCycle;
+                }
             } else if (pickCurr) {
```

Comment:
Basically,  if I remove the earliest drip which is stored in `nextReceivedCycle`. There is nothing we can do because we don't track the next one in the row.